R/pkratio-datamapping.R

```diff
@@ -73,11 +129,71 @@ getDefaultCaptions <- function(data, metaData, variableList = colnames(data)) {
 
 asLegendCaptionSubset <- function(data, metaData) {
   CaptionSubset <- as.character(data)
-
+  
   # If numeric create a character as rounded numeric + unit from metadata
   if ("numeric" %in% class(data)) {
     CaptionSubset <- paste(as.character(round(data)), metaData$Unit, sep = "")
   }
-
+  
   return(CaptionSubset)
 }
+
+
+
+
+check_if_not_numeric <- function(vec,msg="Input must be numeric."){
+  if (!is.numeric(vec)){
+    stop(msg)
+  }
+} 
+
+
+addCaptionsColumn <- function(df,dfinp,newColName = NULL){

```

Comment:
This is a very complicated method here, with many different variables, loop, ifs etc.. 

Because of it's complexity, I would like to 
* First have tests that actually document the behavior
* Second: See if this can be rewritten, simplified. Thinking about whomever is going to have to update the code in 6 months :)